tests/ui-toml/pub_crate_missing_docs/pub_crate_missing_doc.rs

```diff
@@ -0,0 +1,51 @@
+//! this is crate
+#![warn(clippy::missing_docs_in_private_items)]
+
+/// this is mod
+mod my_mod {
+    /// some docs
+    fn priv_with_docs() {}
+    fn priv_no_docs() {}
+    /// some docs
+    pub(crate) fn crate_with_docs() {}
+    pub(crate) fn crate_no_docs() {}
+    /// some docs
+    pub(super) fn super_with_docs() {}
+    pub(super) fn super_no_docs() {}
+
+    mod my_sub {
+        /// some docs
+        fn sub_priv_with_docs() {}
+        fn sub_priv_no_docs() {}
+        /// some docs
+        pub(crate) fn sub_crate_with_docs() {}
+        pub(crate) fn sub_crate_no_docs() {}
+        /// some docs
+        pub(super) fn sub_super_with_docs() {}
+        pub(super) fn sub_super_no_docs() {}
+    }
+
+    /// some docs
+    pub(crate) struct CrateStructWithDocs {
+        /// some docs
+        pub(crate) crate_field_with_docs: (),
+        pub(crate) crate_field_no_docs: (),
+        /// some docs
+        priv_field_with_docs: (),
+        priv_field_no_docs: (),
+    }
+
+    pub(crate) struct CrateStructNoDocs {
+        /// some docs
+        pub(crate) crate_field_with_docs: (),
+        pub(crate) crate_field_no_docs: (),
+        /// some docs
+        priv_field_with_docs: (),
+        priv_field_no_docs: (),
+    }
+}
+
```

Comment:
question: This configuration says `Whether to **only** check for missing documentation in pub(crate) items`. Should the case below in top-level module be checked?
```suggestion
type Typedef = String;
pub type PubTypedef = String;

```

Reply:
I have mixed feelings about this. On one hand `Typedef` is visible inside of the whole crate so it would be checked. However, some people might want to just check stuff that's marked explicitly as `pub(crate)` or `pub(super)` (where `super == crate`).

Reply:
IMO it's okay that `Typedef` is checked, but it seems to be confusing the note of configuration, for example how about the following:
`Whether to check for missing documentation in items visible within the current crate, for example pub(crate) items`

And can the tests like https://github.com/rust-lang/rust-clippy/pull/10303/files#r1106516655 be added?

Reply:
sure I'll do both and ping you again